From 3f5575cfe4016f2d8a34c6e506146b962f7aa083 Mon Sep 17 00:00:00 2001
From: Ronald Wampler <rdwampler@gmail.com>
Date: Sat, 19 Sep 2015 10:14:07 -0400
Subject: [PATCH] Directly convert the URL received from NSOpenPanel to a path
Upstream-Status: Backport

We call `fileURL` here but the method doesn't actually exists (it's actually a property with the getter `isFileURL`). The URLs we obtain from NSOpenPanel are paths so it's probably safe to get the path without checking if it's a proper file URL.
---
 Frameworks/HTMLOutput/src/browser/HOWebViewDelegateHelper.mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Frameworks/HTMLOutput/src/browser/HOWebViewDelegateHelper.mm b/Frameworks/HTMLOutput/src/browser/HOWebViewDelegateHelper.mm
index 17d2cc6..60e5c55 100644
--- ./Frameworks/HTMLOutput/src/browser/HOWebViewDelegateHelper.mm
+++ ./Frameworks/HTMLOutput/src/browser/HOWebViewDelegateHelper.mm
@@ -44,7 +44,7 @@ - (void)webView:(WebView*)sender runOpenPanelForFileButtonWithResultListener:(id
 	NSOpenPanel* panel = [NSOpenPanel openPanel];
 	[panel setDirectoryURL:[NSURL fileURLWithPath:NSHomeDirectory()]];
 	if([panel runModal] == NSOKButton)
-		[resultListener chooseFilename:[[[[panel URLs] objectAtIndex:0] fileURL] path]];
+		[resultListener chooseFilename:[[[panel URLs] objectAtIndex:0] path]];
 }
 
 - (WebView*)webView:(WebView*)sender createWebViewWithRequest:(NSURLRequest*)request
